src/wallet/mod.rs

```diff
@@ -595,30 +592,29 @@ where
             })
             .collect::<Result<Vec<_>, _>>()?;
 
-        let builder_extra_utxos = builder.utxos.as_ref().map(|utxos| {
-            utxos
-                .iter()
-                .filter(|utxo| {
-                    !original_txin
-                        .iter()
-                        .any(|txin| &&txin.previous_output == utxo)
-                })
-                .cloned()
-                .collect()
-        });
-        let (available_utxos, use_all_utxos) = self.get_available_utxos(
+        let builder_extra_utxos = builder
+            .utxos
+            .iter()
+            .filter(|utxo| {
+                !original_txin
+                    .iter()
+                    .any(|txin| &&txin.previous_output == utxo)
+            })
+            .cloned()
+            .collect::<Vec<_>>();
+
+        let (must_use_utxos, may_use_utxos) = self.get_must_may_use_utxos(
             builder.change_policy,
-            &builder_extra_utxos,
             &builder.unspendable,
-            false,
+            &builder_extra_utxos[..],
+            false, // when doing bump_fee `send_all` does not mean use all available utxos
+            builder.manually_selected_only,
         )?;
-        let available_utxos =
-            rbf::filter_available(self.database.borrow().deref(), available_utxos.into_iter())?;
 
-        let (mut must_use_utxos, may_use_utxos) = match use_all_utxos {
-            true => (available_utxos, vec![]),
-            false => (vec![], available_utxos),
-        };
+        let mut must_use_utxos =
+            rbf::filter_available(self.database.borrow().deref(), must_use_utxos.into_iter())?;
+        let may_use_utxos =
+            rbf::filter_available(self.database.borrow().deref(), may_use_utxos.into_iter())?;
```

Comment:
Yeah, I think it should be possible to move it in there and only do one filtering before the utxos are split between "may" and "must" use

Reply:
Ah no, wait a second: you should only do the filtering on `bump_fee`, not `create_tx`. So maybe moving it in there is not a good idea, because then you need an extra parameter to tell whether you are creating or bumping a tx.